src/contextkit-subscriber/property.cpp

```diff
@@ -189,7 +189,9 @@ class UnsubscribeRequest : public Event
         , key_(key)
         , done_(std::move(done))
     {}
-    virtual ~UnsubscribeRequest() {}
+    virtual ~UnsubscribeRequest() {
+        done_.set_value();
+    }
 
     UnsubscribeRequest(SubscribeRequest const&) = delete;
```

Comment:
By 'that' I meant the foo(bar&)=delete syntax. The rest is as clear you your code can possibly be. Anyway, we can take it offline.